cnf-certification-test/lifecycle/podrecreation/podrecreation.go

```diff
@@ -75,33 +78,89 @@ func CountPodsWithDelete(nodeName string, isDelete bool) (count int, err error)
 		return 0, err
 	}
 	count = 0
+	var wg sync.WaitGroup
 	for idx := range pods.Items {
-		for _, or := range pods.Items[idx].OwnerReferences {
-			if or.Kind == DaemonSetString {
-				continue
-			}
-			count++
-			if !isDelete {
-				continue
-			}
-			logrus.Tracef("deleting %s", pods.Items[idx].String())
-			deleteOptions := metav1.DeleteOptions{}
-			gracePeriodSeconds := int64(DefaultGracePeriodInSeconds + time.Duration(*pods.Items[idx].Spec.TerminationGracePeriodSeconds))
-			deleteOptions.GracePeriodSeconds = &gracePeriodSeconds
-
-			err = clients.K8sClient.CoreV1().Pods(pods.Items[idx].Namespace).Delete(context.TODO(), pods.Items[idx].Name, deleteOptions)
-			if err != nil {
-				logrus.Errorf("error deleting %s err: %v", pods.Items[idx].String(), err)
-				return 0, err
-			}
+		if skipDaemonPod(&pods.Items[idx]) {
+			continue
+		}
+		count++
+		if !isDelete {
+			continue
+		}
+		err := deletePod(&pods.Items[idx], waitForDeletion, &wg)
+		if err != nil {
+			logrus.Errorf("error deleting %s", &pods.Items[idx])
 		}
 	}
+
+	wg.Wait()
 	return count, nil
 }
+func skipDaemonPod(pod *v1.Pod) bool {
+	for _, or := range pod.OwnerReferences {
+		if or.Kind == DaemonSetString {
+			return true
+		}
+	}
+	return false
+}
+
+func deletePod(pod *v1.Pod, waitForDeletion bool, wg *sync.WaitGroup) error {
+	clients := clientsholder.GetClientsHolder()
+	logrus.Tracef("deleting %s", pod.String())
+	deleteOptions := metav1.DeleteOptions{}
+	gracePeriodSeconds := int64(DefaultGracePeriodInSeconds + time.Duration(*pod.Spec.TerminationGracePeriodSeconds))
```

Comment:
Maybe I missed this one from previous reviews but, why are you adding this DefaultGracePeriodInSeconds to the  pod's one? I believe doing this should be enough:
```
gracePeriodSeconds := time.Seconds * time.Duration(*pod.Spec.TerminationGracePeriodSeconds)
```